x/perpetual/keeper/updated_leverage.go

```diff
@@ -0,0 +1,48 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	assetprofiletypes "github.com/elys-network/elys/x/assetprofile/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+	"github.com/elys-network/elys/x/perpetual/types"
+)
+
+func (k Keeper) UpdatedLeverage(ctx sdk.Context, mtp types.MTP) (sdk.Dec, error) {
+
+	entry, found := k.assetProfileKeeper.GetEntry(ctx, ptypes.BaseCurrency)
+	if !found {
+		return sdk.ZeroDec(), errorsmod.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+	}
+	baseCurrency := entry.Denom
+
+	priceBaseCurrency := k.oracleKeeper.GetAssetPriceFromDenom(ctx, baseCurrency)
+	if priceBaseCurrency.IsZero() {
+		return sdk.ZeroDec(), fmt.Errorf("token price not set: %s", baseCurrency)
+	}
+
+	custodyInUsdc := mtp.Custody.ToLegacyDec()
+	if mtp.CustodyAsset != baseCurrency {
+		priceCustodyAsset := k.oracleKeeper.GetAssetPriceFromDenom(ctx, mtp.CustodyAsset)
+		if priceBaseCurrency.IsZero() {
+			return sdk.ZeroDec(), fmt.Errorf("token price not set: %s", mtp.CustodyAsset)
+		}
+		custodyInUsdc = mtp.Custody.ToLegacyDec().Mul(priceCustodyAsset.Quo(priceBaseCurrency))
+	}
+	denominator := custodyInUsdc.Sub(mtp.Liabilities.ToLegacyDec())
```

Comment:
Incorrect. For short, liabilities will be trading in asset, can't subtract different units